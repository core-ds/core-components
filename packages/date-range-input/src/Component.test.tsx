import React from 'react';
import userEvent from '@testing-library/user-event';
import { render, waitFor } from '@testing-library/react';

import { DateRangeInput } from './index';

describe('DateRangeInput', () => {
    describe('Display tests', () => {
        it('should match snapshot', () => {
            expect(
                render(<DateRangeInput defaultValue='12.12.2021 - 10.04.2022' />).container,
            ).toMatchSnapshot();
        });
    });

    it('should set `data-test-id` attribute', () => {
        const testId = 'test-id';
        const { getByTestId } = render(<DateRangeInput dataTestId={testId} />);

        expect(getByTestId(testId)).toBeInTheDocument();
    });

    it('should set custom class', () => {
        const className = 'custom-class';
        const { container } = render(<DateRangeInput className={className} />);

        expect(container.firstElementChild).toHaveClass(className);
    });

    describe('Uncontrolled-way', () => {
        it('should set default value to input', () => {
            const value = '12.12.2021 - 10.04.2022';
            const { queryByRole } = render(<DateRangeInput defaultValue={value} />);

            expect(queryByRole('textbox')).toHaveValue(value);
        });

        it('should set value to input', async () => {
            const value = '12.12.2021 - 10.04.2022';
            const { queryByRole } = render(<DateRangeInput />);

            const input = queryByRole('textbox') as HTMLInputElement;

            await userEvent.type(input, value);

            await waitFor(() => {
                expect(input).toHaveValue(value);
            });
        });
    });

    describe('Callback tests', () => {
        it('should call onChange callback', async () => {
            const onChange = jest.fn();
            const onComplete = jest.fn();
            const value = '12.12.2021 - 10.04.2022';
            const { queryByRole } = render(
                <DateRangeInput onChange={onChange} onComplete={onComplete} />,
            );

            const input = queryByRole('textbox') as HTMLInputElement;

            await userEvent.type(input, value);

            await waitFor(() => {
                expect(onComplete).toBeCalledTimes(2);
                expect(onChange).toBeCalledTimes(value.length + 2);
            });
        });
    });

    describe('Render tests', () => {
        test('should unmount without errors', () => {
            const { unmount } = render(<DateRangeInput />);

            expect(unmount).not.toThrowError();
        });
    });
});
