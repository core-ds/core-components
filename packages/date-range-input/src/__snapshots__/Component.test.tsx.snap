// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangeInput Display tests should match snapshot 1`] = `
<div>
  <div
    class="component"
  >
    <div
      class="component component s hasRightAddons"
    >
      <div
        class="inner inner filled"
      >
        <div
          class="inputWrapper"
        >
          <div
            class="input"
          >
            <input
              class="input input"
              type="text"
              value="12.12.2021 - 10.04.2022"
            />
          </div>
        </div>
        <div
          class="addons rightAddons"
        />
      </div>
    </div>
  </div>
</div>
`;
